packages/start/config/index.js

```diff
@@ -0,0 +1,179 @@
+import { serverFunctions } from "@vinxi/server-functions/plugin";
+import { serverTransform } from "@vinxi/server-functions/server";
+import defu from "defu";
+import { join } from "node:path";
+import { fileURLToPath } from "node:url";
+import { createApp, resolve } from "vinxi";
+import { normalize } from "vinxi/lib/path";
+import { config } from "vinxi/plugins/config";
+import solid from "vite-plugin-solid";
+import { SolidStartClientFileRouter, SolidStartServerFileRouter } from "./fs-router.js";
+import { serverComponents } from "./server-components.js";
+
+const DEFAULT_EXTENSIONS = ["js", "jsx", "ts", "tsx"];
+
+function solidStartClientFsRouter(config) {
+  return (router, app) =>
+    new SolidStartClientFileRouter(
+      {
+        dir: resolve.absolute(config.dir, router.root),
+        extensions: config.extensions ?? ["js", "jsx", "ts", "tsx"]
+      },
+      router,
+      app
+    );
+}
+
+function solidStartServerFsRouter(config) {
+  return (router, app) =>
+    new SolidStartServerFileRouter(
+      {
+        dir: resolve.absolute(config.dir, router.root),
+        extensions: config.extensions ?? ["js", "jsx", "ts", "tsx"]
+      },
+      router,
+      app
+    );
+}
+
+export function defineConfig(baseConfig = {}) {
```

Comment:
Thoughts on converting this file to typescript and doing something like this? This would allow the config to be typed

```
import { InlineConfig } from "vite";

type SolidStartInlineConfig = Omit<InlineConfig, "router"> & {
  start?: {
    ssr?: boolean,
    extensions?: string[],
    server?: any,
    appRoot?: string,
    middleware?: string
    islands?: boolean
  }
}

export function defineConfig(baseConfig: SolidStartInlineConfig = {}) {
```

Reply:
If it works. I'm unclear what Vite transforms in these scenarios. Whether it is part of the build or part of the runtime. We might be better to JSDocs it etc..